packages/react/src/components/header/components/navigationLanguageSelector/NavigationLanguageSelector.tsx

```diff
@@ -2,67 +2,91 @@ import React, { PropsWithChildren, useEffect } from 'react';
 
 import {
   LanguageOption,
+  useAvailableLanguages,
   useActiveLanguage,
   useSetAvailableLanguages,
   useSetLanguage,
 } from '../../../../context/languageContext';
 import classNames from '../../../../utils/classNames';
+import { getChildElementsEvenIfContainersInbetween } from '../../../../utils/getChildren';
 import { withDefaultPrevented } from '../../../../utils/useCallback';
 import { HeaderActionBarItemWithDropdown } from '../headerActionBarItem';
 import { IconAngleDown, IconAngleUp, IconGlobe } from '../../../../icons';
 import classes from './NavigationLanguageSelector.module.scss';
-import { useHeaderContext } from '../../HeaderContext';
+import { useHeaderContext, useSetHeaderContext } from '../../HeaderContext';
 
 export type LanguageSelectorProps = PropsWithChildren<{
   /**
    * Array of languages as LanguageOption objects.
    */
-  languages: LanguageOption[];
+  languages?: LanguageOption[];
+  /**
+   * Mobile mode.
+   */
+  fullWidthForMobile?: boolean;
```

Comment:
This could be an internal prop so it's not shown outside to users because they don't have a need for it. Just slap `@internal` in this here prop docs and that's it 👍